File: check-sessions.ts
import postgres from "postgres";
import { config } from "dotenv";

config({ path: "D:\\Whiteroom\\.env" });

async function run() {
  const url = process.env.DATABASE_URL!;
  console.log("Connecting to:", url);
  const client = postgres(url);
  try {
    const sessions = await client`
      SELECT id, phone, verified, expires_at, created_at
      FROM whatsapp_sessions
      ORDER BY created_at DESC
      LIMIT 10;
    `;
    console.log("Latest 10 WhatsApp sessions:");
    for (const s of sessions) {
      console.log(`ID: ${s.id} | Phone: ${s.phone} | Verified: ${s.verified} | ExpiresAt: ${s.expires_at} | CreatedAt: ${s.created_at}`);
    }
  } catch (err) {
    console.error("Error connecting to DB:", err);
  } finally {
    await client.end();
  }
}

run();

File: check_sessions.py
import os
import sys
from dotenv import load_dotenv

# Load .env
load_dotenv("D:\\Whiteroom\\.env")

url = os.environ.get("DATABASE_URL")
print("Connecting to DB:", url)

import psycopg2
try:
    conn = psycopg2.connect(url)
    cur = conn.cursor()
    
    # Get latest 10 whatsapp sessions
    cur.execute("""
        SELECT id, phone, verified, expires_at, created_at
        FROM whatsapp_sessions
        ORDER BY created_at DESC
        LIMIT 10;
    """)
    sessions = cur.fetchall()
    print("Latest 10 WhatsApp sessions:")
    for s in sessions:
        print(f"ID: {s[0]} | Phone: {s[1]} | Verified: {s[2]} | ExpiresAt: {s[3]} | CreatedAt: {s[4]}")
        
    cur.close()
    conn.close()
except Exception as e:
    print("Error:", e)
